web/app/src/ui/accounts/AuthenticationRequired.tsx

```
@@ -0,0 +1,39 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { getAuthenticationState } from "../../storage/authentication";
+import { UserPlus } from "@phosphor-icons/react";
+import { LoginForm } from "./LoginForm";
+
+export function AuthenticationRequired({
+	message,
+	children,
+}: React.PropsWithChildren<{
+	message?: string;
+}>) {
+	const authState = getAuthenticationState();
```

We usually have a `useAuthenticatedUser` hook for this

---

Do you mean that `useAuthenticatedUser` (that I named `useAuthenticatedAccount` as I used the word "account" in the rest of the app) should throw an exception then handled by the error boundary instead of this component?